src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -2964,9 +2963,21 @@ private RefKind GetEffectiveParameterRefKind(ParameterSymbol parameter, RefKind
                 }
                 else
                 {
+                    RefKind argumentRefKind = arguments.RefKind(argumentPosition);
                     RefKind parameterRefKind = parameters.ParameterRefKinds.IsDefault ? RefKind.None : parameters.ParameterRefKinds[argumentPosition];
                     bool forExtensionMethodThisArg = arguments.IsExtensionMethodThisArgument(argumentPosition);
 
+                    if (forExtensionMethodThisArg)
+                    {
+                        Debug.Assert(argumentRefKind == RefKind.None);
+                        if (parameterRefKind == RefKind.Ref)
```

Comment:
For the purposes of overload resolution, ref-readonly is treated as none.

---
In reply to: [122015601](https://github.com/dotnet/roslyn/pull/19703#discussion_r122015601) [](ancestors = 122015601)